packages/components/src/components/Badge/index.tsx

```diff
@@ -6,28 +6,38 @@ import { Text } from '../Text';
 export interface BadgeProps {
   color?: 'accent' | 'neutral';
   icon?: IconNames;
+  isPadded?: boolean;
 }
 
 export const Badge: React.FC<BadgeProps> = ({
   color = 'neutral',
   icon,
+  isPadded,
   children,
-}) => (
-  <Stack
-    css={{
-      alignItems: 'center',
-      padding: icon ? '4px 8px' : '1px 6px',
-      userSelect: 'none',
+}) => {
+  let padding = '1px 6px';
 
-      borderRadius: '999px',
+  if (isPadded) {
```

Comment:
does it make sense to keep the logic that if it has an icon, it should be padded? so `isPadded || icon`?

Reply:
The values for `isPadded`, `icon` and default are all different. Made sure that the `Badge` stays the same for previous implementations.